Add anonymous project id to telemetry

rijkvanzanten opened the pull request:
Allows us to anonymously differentiate between telemetry records

rijkvanzanten committed:
Install uuid

rijkvanzanten committed:
Add project id on migrate/install

rijkvanzanten committed:
Include project id in telemetry

rijkvanzanten committed:
Add changeset

connorwinston commented on api/src/database/migrations/20250613A-add-project-id.ts:
```diff
@@ -0,0 +1,26 @@
+
+	if (existing) {
+		await knex('directus_settings').update({
+			project_id: uuid(),
```
We should get the timestamp of the first database migration

```
const timestamp = await knex('directus_migrations').select('timestamp').first();
``` 

and use that when creating the uuid like 

```
uuid({ msecs: 12345 }) //  RFC "timestamp" field (Number of milliseconds, unix epoch)
```

connorwinston reviewed (dismissed):
Just one change, otherwise LGTM!

licitdev committed:
Fix snapshot

licitdev reviewed (commented):
Should `project_id` be added to the [types package](https://github.com/directus/directus/blob/2c8cb0e13b6558cde64b605f21b7931341e51f8b/packages/types/src/settings.ts) and the sdk?

connorwinston committed:
Merge branch 'main' into rijk/now-996

connorwinston committed:
Use migration timestamp for project_id UUID generation

Updated the migration to use the timestamp from 'directus_migrations' as the msecs value when generating the project_id UUID.

connorwinston commented:
@rijkvanzanten I added to the migration generating the UUID v7 from the first migrations timestamp, how does it look?

licitdev committed:
Fix formatting

licitdev committed:
Add into types and sdk

ComfortablyCoding committed:
Merge branch 'main' into rijk/now-996

ComfortablyCoding committed:
update changeset
